File: app/src/main/java/org/kindone/willingtodo/pomodorocontrol/ControlStopped.java
package org.kindone.willingtodo.pomodorocontrol;

/**
 * Created by kindone on 2017. 1. 2..
 */

public class ControlStopped extends ControlState {
    public ControlStopped(ControlContext context) {
        super(context);
    }

    @Override
    public void start() {
        changeState(new ControlRunning(mContext));
    }

    @Override
    public void pause() {
        // DO NOTHING
    }

    @Override
    public void resume() {
        // DO NOTHING
    }

    @Override
    public void stop() {
        // DO NOTHING
    }
}

File: app/src/main/java/org/kindone/willingtodo/pomodorocontrol/ControlStopped.kt
package org.kindone.willingtodo.pomodorocontrol

/**
 * Created by kindone on 2017. 1. 2..
 */

class ControlStopped(context: ControlContext) : ControlState(context) {

    override fun start() {
        changeState(ControlRunning(mContext))
    }

    override fun pause() {
        // DO NOTHING
    }

    override fun resume() {
        // DO NOTHING
    }

    override fun stop() {
        // DO NOTHING
    }

    override fun getMode(): Int {
        return ControlState.MODE_PLAYING
    }
}
